server/src/com/thoughtworks/go/server/service/EnvironmentConfigService.java

```diff
@@ -172,12 +174,12 @@ public void performEdit() {
                     result.conflict(LocalizedMessage.string("CANNOT_ADD_ENV_ALREADY_EXISTS", config.name()));
                     return;
                 }
-                goConfigService.addEnvironment(config);
+                goConfigService.addEnvironment(config, user);
```

Comment:
Consider using the AddEnvironmentCommand for consistency as you use UpdateEnvironmentCommand and DeleteEnvrironmentCommand in the respective functions